proto/cacheclient.proto

```diff
@@ -459,7 +459,8 @@ message _SortedSetFetchRequest {
   message _All {}
 
   message _Limit {
-    uint32 limit = 1;
+    uint32 limit = 1;   // aka count; number of elements to return
```

Comment:
Is there a reason why we are doing a range in here instead of the 

```
  oneof range {
    _ByIndex by_index = 5;
    _ByScore by_score = 6;
  }
```
It feels more natural to query the range.

Reply:
Sorted sets doing query by index by default, and hence what "limit" does is essentially same as the range endpoints. 
In redis, you can do
- `ZRANGE key start stop`, e.g. `ZRANGE myzset 5 100` gives you index between 5 and 100 inclusive
- `ZRANGEBYSCORE key start stop LIMIT offset count`, e.g.  `ZRANGEBYSCORE myzset -inf 1000 LIMIT 5 96`, gives you all elements from the smallest up to score == 1000, and only returns from between 5 and 100 among the range of elements.  
`ZRANGE` command does not have "limit" option since it can be achieved by the start and stop indexes, but for `ZRANGEBYSCORE`, "limit" works on top of start and stop score endpoints. 
In other words the limit in the proto indeed determine the number of results returned.  Internally in ms2, this limit thing is mapped to the start and stop for ZRANGE, and mapped to count and offset for ZRANGEBYSCORE.  

Only for fetching by score we had this additional option to pick the start and stop scores.  For the already implemented fetching by index, there's no additional options that one can pick.

Reply:
Not sure if I answered your question.. IMO `limit` is something universal, so it's fine to stay outside of the "range" field. For fetch-by-index, there's no other arguments one can provide, but for other fetch-by types, there can be additional arguments (for example, when fetch-by-lex, one need to give min and max item values for the range).

Reply:
Just summarizing what you expressed, limit allows to restrict and offset the result set. When ByIndex is specified this allows us to specify offset and limits. This option can also apply to restrict the number of results when we get a range by score.